packages/spark/base/inputs/_input-container.scss

```diff
@@ -11,3 +11,248 @@
 .sprk-b-InputContainer__visibility-toggle {
   margin: $sprk-visibility-control-margin;
 }
+
+// Applies to any huge input
+.sprk-b-InputContainer--huge .sprk-b-TextInput,
+.sprk-b-InputContainer--huge .sprk-b-Select {
+  position: relative;
+  transition: $sprk-input-huge-transition;
+  transition-delay: $sprk-input-huge-transition-delay;
+  height: $sprk-input-huge-height;
+  border-width: $sprk-input-huge-border-width;
+  border-radius: $sprk-text-input-border-radius;
+  font-size: $sprk-input-huge-font-size;
+  padding: $sprk-input-huge-padding;
+  box-shadow: $sprk-input-huge-box-shadow;
+}
+
+// Adjust for select open icon
+.sprk-b-InputContainer--huge .sprk-b-Select {
+  padding-right: $sprk-select-input-huge-padding-right;
+}
+
+.sprk-b-InputContainer--huge .sprk-b-TextInput:focus,
+.sprk-b-InputContainer--huge .sprk-b-Select:focus {
+  border-width: $sprk-input-huge-focus-border-width;
+}
+
+/*
+* TODO as part of deprecate task:
+* delete this selector `.sprk-b-TextInput--float-label`
+*/
+
+.sprk-b-InputContainer--huge .sprk-b-TextInput::placeholder,
+.sprk-b-InputContainer--huge .sprk-b-TextInput.sprk-b-TextInput--float-label::placeholder,
+.sprk-b-InputContainer--huge .sprk-b-TextInput.sprk-b-Input--has-floating-label::placeholder,
+.sprk-b-InputContainer--huge .sprk-b-Select.sprk-b-Input--has-floating-label {
+  font-size: $sprk-input-huge-font-size;
+}
+
+.sprk-b-InputContainer--huge .sprk-b-TextInput:-ms-input-placeholder,
+.sprk-b-InputContainer--huge .sprk-b-TextInput.sprk-b-TextInput--float-label:-ms-input-placeholder,
+.sprk-b-InputContainer--huge .sprk-b-TextInput.sprk-b-Input--has-floating-label:-ms-input-placeholder {
+  font-size: $sprk-input-huge-font-size;
+}
+
+// Applies when input is first direct child of the container div
+// to create floating label effect
+// Label element must follow the input element
+
+// Huge styles for Monetary Input
+.sprk-b-InputContainer--huge .sprk-b-TextInputIconContainer .sprk-b-TextInput {
+  padding: $sprk-text-input-has-icon-padding;
+}
+
+// Takes in account the height of the select to position select icon
+.sprk-b-InputContainer--huge .sprk-b-SelectContainer__icon {
+  margin-top: $sprk-select-input-huge-icon-margin-top;
+}
+
+.sprk-b-InputContainer--huge .sprk-b-TextInputIconContainer :first-child.sprk-b-TextInput {
+  padding: $sprk-text-input-huge-has-text-icon-padding;
```

Comment:
Philosophical question for later (don't change anything now)...but shouldn't these padding and margin values be font-size relative? We don't have design direction for that explicitly, but could be what we want in the future. Also, fewer things for a theme developer to need to specify.

Reply:
Yes, they should be haha. Now, I know, and I can't unknow...*gollum voice*  MUST FIX